### PR TITLE
DPC-820: As a user, I want to add my public keys for API authentication.

### PR DESCRIPTION
**Pull request by switzersc-usds**
Waiting on #363 -- This PR builds on that and includes a lot of those commits. Will rebase when 363 is merged.

**Why**

Users need to supply their public keys via the dashboard so that they can authenticate with the API.

**What Changed**

* Adds public key creation UI
* Adds list of public keys to dashboard

See screenshots in comment below.

**Choices Made**

Follows pattern established by client token features (#363). Could be DRYer but we're moving quickly to get to launch.

**Tickets closed**:

Give a list of tickets closed in this PR.

### CODE REVIEW

File: dpc-web/app/models/registered_organization.rb [hunk 1]
```diff
@@ -14,4 +14,9 @@ def client_tokens
     # [{'id' => '4r85cfb4-dc36-4cd0-b8f8-400a6dea2d66', 'label' => 'Test Token 1', 'createdAt' => Time.now.iso8601, 'expiresAt' => Time.now.iso8601}]
     ClientTokenManager.new(api_env: api_env, organization: organization).client_tokens
   end
+
+  def public_keys
+    # [{'id' => '4r85cfb4-dc36-4cd0-b8f8-400a6dea2d66', 'label' => 'Test Token 1', 'createdAt' => Time.now.iso8601, 'expiresAt' => Time.now.iso8601}]
+    PublicKeyManager.new(api_env: api_env, organization: organization).public_keys
```
Comment (switzersc-usds) [inline dpc-web/app/models/registered_organization.rb]: This is not an array either

**Comment by jaredcunha-usds** [comment]
Null states for tokens and keys

Screen Shot 2019-11-13 at 2 41 52 PM

Add new public key
Screen Shot 2019-11-13 at 2 37 50 PM

Dashboard with tokens and keys
Screen Shot 2019-11-13 at 2 37 33 PM

Client tokens show
Screen Shot 2019-11-13 at 2 45 00 PM

File: dpc-web/app/views/client_tokens/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      <div class="ds-c-alert ds-u-margin-bottom--7">
+        <div class="ds-c-alert__body">
+          <h2 class="ds-c-alert__heading">Copy or download your token right now! You won’t be able to see again.</h2>
+        </div>
+      </div>
+      <h2 class="box__heading ds-u-margin-bottom--0"><%= @client_token['label'] %></h2>
+      <p class="ds-u-color--muted ds-u-margin-top--0">Created at <%= formatted_datestr(@client_token['createdAt']) %></p>
+      <div class="ds-u-margin-y--5">
+        <label for="token" class="ds-c-label ds-u-font-size--h3">Your token</label>
+          <textarea id="token" class="ds-c-field ds-c-field--copy" rows="5" readonly="readonly"><%= @client_token['token'] %></textarea>
```
Comment (jaredcunha-usds) [inline dpc-web/app/views/client_tokens/show.html.erb]: @switzersc-usds feel free to add to the rows on this. I wasn't sure how to tall to make it.

Comment (switzersc-usds) [reply dpc-web/app/views/client_tokens/show.html.erb]: thanks! changed it to 9

File: dpc-web/app/services/public_key_manager.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      []
+    end
+  end
+
+  def registered_org
+    @registered_org ||= organization.registered_organizations.find_by(api_env: api_env)
+  end
+
+  def strip_carriage_returns(str)
+    str.gsub(/\r/, '')
```
[the remark this answers is not in the document]
Comment (switzersc-usds) [reply dpc-web/app/services/public_key_manager.rb]: `gsub` is the only method that worked with the regex substitution I'm doing here

**Comment by switzersc-usds** [comment]
Not fixing the 1 code climate issue because it's not a good suggestion. Changing it to the method it suggests does not achieve the same functionality (and the test fails).

**Comment by switzersc-usds** [comment]
Talked IRL about error messaging: This isn't currently implemented (we do not pass API errors back to users) but will once the API error messaging is standardized.

**Comment by ronaldheft-usds** [comment]
@switzersc-usds Is this ready to merge once we have golden macaroons in DEV? Planning on rolling out golden macaroons in DEV today. Going to be at JP around 2pm if you want to sync up on open sandbox.

**Comment by ronaldheft-usds** [comment]
Updating this PR to latest master and will be deploying this branch out in DEV with https://github.com/CMSgov/dpc-ops/pull/84 to verify the golden macaroon is functioning correctly.

**Pull request merged**